edgelet/edgelet-core/src/settings.rs

```diff
@@ -290,24 +314,130 @@ impl Listen {
     }
 }
 
+#[derive(Clone, Debug, Fail)]
+pub enum CertificateConfigError {
+    #[fail(
+        display = "URI {} is unsupported for '{}'. Please check the config.yaml file.",
+        _0, _1
+    )]
+    UnsupportedUri(String, &'static str),
+
+    #[fail(
+        display = "File URI {} is unsupported for '{}'. Please check the config.yaml file.",
+        _0, _1
+    )]
+    UnsupportedFileUri(String, &'static str),
+
+    #[fail(
+        display = "Error parsing URI {} specified for '{}'. Please check the config.yaml file.",
+        _0, _1
+    )]
+    InvalidUri(String, &'static str),
+
+    #[fail(
+        display = "Invalid file URI {} path specified for '{}'. Please check the config.yaml file.",
+        _0, _1
+    )]
+    InvalidUriFilePath(String, &'static str),
+}
+
 #[derive(Clone, Debug, serde_derive::Deserialize, serde_derive::Serialize)]
 pub struct Certificates {
-    device_ca_cert: PathBuf,
-    device_ca_pk: PathBuf,
-    trusted_ca_certs: PathBuf,
+    device_ca_cert: String,
+    device_ca_pk: String,
+    trusted_ca_certs: String,
```

Comment:
Can we move this to Url too and convert PathBufs to Urls with the file scheme? Seems like we are downgrading the type from a more full-featured type to a string.

Reply:
serde URL will fail if a user passed in a path. I am all in for making this consistent but I don't know how.

Reply:
Because the value supports both a `PathBuf` and a `Url`, I'm not sure we can deserialize it from the config file.

Also note that even if it was a `PathBuf`, serde does not support serializing / deserializing `PathBuf` that are not also valid utf-8 anyway. So anything we could parse with `PathBuf` before, we'll still be able to parse with `String`.

Reply:
Per our convo, I think we can leave these as is.